atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheImpl.java

```diff
@@ -150,10 +153,6 @@ synchronized LockWatchEventCacheState getStateForTesting() {
                 .build();
     }
 
-    private synchronized void removeFromTimestampState(long startTimestamp) {
```

Comment:
This looks safe, all the introduced race conditions are either OK because the calls for the same timestamp are guaranteed to be executed serially, it's a race condition with `.clear()` which is fine, or potentially slightly delayed retention